### PR TITLE
Using Jump servers with SSH 

### PR DESCRIPTION
**Pull request by Chen2x**
New blog post by me. Please take a look to ensure I have done everything I need.

### CODE REVIEW

File: _posts/chen_chen/2017-06-22-Using-Jump-Servers-in-SSH.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+~~~
+ 
+Which would then allow chain server hopping like so:
+~~~BASH
+ssh -A user@serverA
+serverA# ssh -A user@serverB
+serverB# ssh user@serverC
+~~~
+## Port Forwarding
+Firewalls can provide a front line defense by restricting access to internal services by connections from the Internet, however, there may be cases where a set of “trusted” users require access to these internal services. In the situation that the firewall grants specific servers access to the internal services, these servers as jump servers to establish a secured tunnel in order to access specific, identifiable resources. This act is called port forwarding. When the user makes a request, it is sent to the jump server which forwards the request to the internal resource on the behalf of the user computer. Once the request is granted, the response is then sent back to the jump server which forwards the response to the user computer through the SSH tunnel.
```
Comment (jmimico) [inline _posts/chen_chen/2017-06-22-Using-Jump-Servers-in-SSH.md]: Possible forgotten word?
Orig: these servers **as** jump servers
'these servers act as jump servers' ?

**Review by jmimico — COMMENTED** [review]
Good post Chen. I found the section about agent forwarding particularly useful. You CAN teach an old dog new tricks!

File: _posts/chen_chen/2017-06-22-Using-Jump-Servers-in-SSH.md (hunk 1 above)
Comment (Chen2x) [reply _posts/chen_chen/2017-06-22-Using-Jump-Servers-in-SSH.md]: Fixed it now! Thanks for the feedback!

**Comment by Chen2x** [comment]
Thank you Jared!

File: _posts/chen_chen/2017-06-22-Using-Jump-Servers-in-SSH.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+</figure>
+
+## Agent Forwarding
+
+When establishing an SSH connection with default parameters, the private key is only kept on the local computer and the actual communication between the server and computer is using a challenge string. 
+ 
+However, since the private key is only stored on the user computer, there is a problem when trying to establish a connection between a jump server to another remote server due to the fact that the jump server does not contain the private key. The solution to avoid storing keys on the jump server and compromising security is using an agent on each connection. Using agent forwarding, multiple servers can be jumped through   
+ 
+Despite the name including the term “forwarding” the actual mechanism for agent forwarding does not move the key at all. What the agent does is act as a redirect for when the private key is challenged. Since the private key must never leave the local computer, the agent forwards the challenge down each level of the SSH connection until it reaches the client local machine. Once the challenge is encrypted by the private key, the response is then forwarded by the key back to the server that issued the challenge.
```
Comment (junjun-zhang) [inline _posts/chen_chen/2017-06-22-Using-Jump-Servers-in-SSH.md]: Should the following reads as '... forwarded by the *agent* back to the server ...'

> the response is then forwarded by the key back to the server that issued the challenge

File: _posts/chen_chen/2017-06-22-Using-Jump-Servers-in-SSH.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+<figure>
+    <img src="{{site.urlimg}}chen_chen/JumpServer.png" />
+    <figcaption>Accessing a network of servers using a jump server</figcaption>
+</figure>
+
+## Agent Forwarding
+
+When establishing an SSH connection with default parameters, the private key is only kept on the local computer and the actual communication between the server and computer is using a challenge string. 
+ 
+However, since the private key is only stored on the user computer, there is a problem when trying to establish a connection between a jump server to another remote server due to the fact that the jump server does not contain the private key. The solution to avoid storing keys on the jump server and compromising security is using an agent on each connection. Using agent forwarding, multiple servers can be jumped through
```
Comment (junjun-zhang) [inline _posts/chen_chen/2017-06-22-Using-Jump-Servers-in-SSH.md]: missing period at the end.

File: _posts/chen_chen/2017-06-22-Using-Jump-Servers-in-SSH.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+ 
+However, since the private key is only stored on the user computer, there is a problem when trying to establish a connection between a jump server to another remote server due to the fact that the jump server does not contain the private key. The solution to avoid storing keys on the jump server and compromising security is using an agent on each connection. Using agent forwarding, multiple servers can be jumped through   
+ 
+Despite the name including the term “forwarding” the actual mechanism for agent forwarding does not move the key at all. What the agent does is act as a redirect for when the private key is challenged. Since the private key must never leave the local computer, the agent forwards the challenge down each level of the SSH connection until it reaches the client local machine. Once the challenge is encrypted by the private key, the response is then forwarded by the key back to the server that issued the challenge.
+
+To agent Forward, just include “-A” after the ssh command to call agent.
```
Comment (junjun-zhang) [inline _posts/chen_chen/2017-06-22-Using-Jump-Servers-in-SSH.md]: lower case: 'forward'?

File: _posts/chen_chen/2017-06-22-Using-Jump-Servers-in-SSH.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+ 
+* Remote port forwarding: Allows the server to access data from a local computer by using its connection. This is useful when a computer connected to the server contains files that users on the server need access to. The local computer forwards the port of which the internal service is stored. This allows users connected to the server to access the specified port on the local computer that forwarded the port.
+ 
+* Dynamic port forwarding: Creates a tunnel for sending and receiving data through firewall pinholes by using multiple ports and servers. It is more complicated to set up than the remote and local port forwarding but it is a very powerful tool that can run several applications from different sources at a time. The source port is not specified since more set up is needed in order to for connections to be made to the specific ports.
+
+<figure>
+    <img src="{{site.urlimg}}chen_chen/PortForwarding.png" />
+    <figcaption>If a local port is forwarded to a webpage, the webpage can be accessed by entering
```
Comment (junjun-zhang) [inline _posts/chen_chen/2017-06-22-Using-Jump-Servers-in-SSH.md]: 'webpage' should be 'web server'

File: _posts/chen_chen/2017-06-22-Using-Jump-Servers-in-SSH.md (hunk 2 above)
Comment (Chen2x) [reply _posts/chen_chen/2017-06-22-Using-Jump-Servers-in-SSH.md]: thank you I fixed it now

File: _posts/chen_chen/2017-06-22-Using-Jump-Servers-in-SSH.md (hunk 3 above)
Comment (Chen2x) [reply _posts/chen_chen/2017-06-22-Using-Jump-Servers-in-SSH.md]: I fixed it now

File: _posts/chen_chen/2017-06-22-Using-Jump-Servers-in-SSH.md (hunk 5 above)
Comment (Chen2x) [reply _posts/chen_chen/2017-06-22-Using-Jump-Servers-in-SSH.md]: changed it now

**Review by Fgerthoffert — APPROVED** [review]

**Pull request merged**